File: ProyectoFinal.Antares.Domain/Modelos/Usuario.cs
using ProyectoFinal.Antares.Domain.Enums;

namespace ProyectoFinal.Antares.Domain.Modelos;

public class Usuario : BaseModel
{
    public string NombreUsuario { get; set; } = null!;
    
    public string Password { get; set; } = null!;

    public TipoUsuario Tipo { get; set; }
    
    public int Dni { get; set; }
    
    public string Direccion { get; set; } = null!;
    
    public string Mail { get; set; } = null!;

    public string Telefono { get; set; } = null!;
}

File: FrontendWeb/ProyectoFinal_Antares_FrontEnd_Web/src/app/Models/usuario.ts
export interface Usuario
{
    id: number;
    nombreUsuario: string;
    password: string;
    tipo: TipoUsuario;
    dni: number;
    mail: string;
    direccion: string;
    telefono: string;
}

export enum TipoUsuario
{
  Admin,
  Mozo,
  Delivery,
  Cliente,
  Caja
}

export const TipoUsuarioMapping: Record<string, TipoUsuario> = {
  "Admin" : TipoUsuario.Admin,
  "Mozo" : TipoUsuario.Mozo,
  "Delivery": TipoUsuario.Delivery,
  "Cliente": TipoUsuario.Cliente,
  "Caja": TipoUsuario.Caja
};
